### PR TITLE
Generate and download xml and pdf server side

### PR DESCRIPTION
**Pull request by danielhusar**
Fixne tieto tri issues:

https://trello.com/c/Z3SmSHPQ/132-ked-je-nahoden%C3%BD-popup-blocker-nechce-stiahnu%C5%A5-s%C3%BAbory-po-pokuse-z-linku-otvori%C5%A5-v-novom-okne-vyhadzuje-chyby-detto-xml-ani-pdf

https://trello.com/c/i5PSzA35/124-testovanie-chyby-pdf-s%C3%BAbor-je-po%C5%A1koden%C3%BD-alebo-pr%C3%A1zdny

https://trello.com/c/B3UrdVyU/130-posielanie-mailu-v-screene-v%C3%BDpo%C4%8Det-dane-za-rok-2019-po-zadan%C3%AD-mailu-na-odoslanie-formul%C3%A1ra-nefunguje-e%C5%A1te-nekomunikuje-so-server

Problem bol v podstate spolocny, data type ked sme poslali request na server sa stratili. Vyriesil som to posielanim `taxFormUserInput` namiesto `taxForm` a prekonvertovanim to spat na `taxForm` na serveri.

JSON.parse 😬 ale zvolil som easy cestu namiesto separatneho inputu pre kazdu hodnotu.

### CODE REVIEW

**Comment by mathio** [comment]
> WIP: Niektore data typy ako napr array, Decimal nepreziju request momentalne

FYI rovnaka chyba je aj [pri posielanie emailu](https://trello.com/c/B3UrdVyU/130-posielanie-mailu-v-screene-v%C3%BDpo%C4%8Det-dane-za-rok-2019-po-zadan%C3%AD-mailu-na-odoslanie-formul%C3%A1ra-nefunguje-e%C5%A1te-nekomunikuje-so-server), tam tiez generujeme XML a PDF.

File: src/pages/api/xml.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+export default async (
+  req: NextApiRequest,
+  res: NextApiResponse,
+): Promise<void> => {
+  if (!req.body.taxFormUserInput) {
+    return res.status(500).json({ error: 'invalid data' })
+  }
+
+  const formInput: TaxFormUserInput = JSON.parse(req.body.taxFormUserInput)
+  const form: TaxForm = calculate(setDate(formInput))
```
Comment (wormyy) [inline src/pages/api/xml.ts:17]:
```suggestion
  const form: TaxForm = calculate(setDate(formInput, new Date()))
```

Tu musi byt toto, lebo `setDate` je zla neintuitivna funkcia a automaticky tam dosadi nejaky datum v minulosti :D Myslim ze som to tak spravil kvoli testom, ale chcelo by to refaktornut

Comment (danielhusar) [reply src/pages/api/xml.ts]: Nemali by sme potom to opravit aj tu?

https://github.com/priznanie-digital/priznanie-digital/blob/master/src/pages/_app.tsx#L44

Comment (wormyy) [reply src/pages/api/xml.ts]: Hej, je to bugnute. https://github.com/priznanie-digital/priznanie-digital/pull/162 tento PR by to mal riesit vsade, takze mozes to nechat tak ako to je teraz

Comment (wormyy) [reply src/pages/api/xml.ts]: Sorry ak si to uz fixol tiez :(

Comment (danielhusar) [reply src/pages/api/xml.ts]: nevadi revertnem to :)

**Pull request merged**